### PR TITLE
Added OutboundShipments.create_fulfillment_order

### PR DESCRIPTION
**Pull request by tgehrs**
Starting with a small PR in order to make sure things are on the right track, can work to fill in some of the other methods for OutboundShipments if needed.

### CODE REVIEW

File: mws/apis/outbound_shipments.py [hunk 1]
```diff
@@ -32,15 +32,50 @@ def get_fulfillment_preview(self):
...
+            "SellerFulfillmentOrderId": seller_fulfillment_order_id,
+            "FulfillmentAction": fulfillment_action,
+            "DisplayableOrderId": displayable_order_id,
+            "DisplayableOrderDateTime": displayable_order_datetime,
+            "DisplayableOrderComment": displayable_order_comment,
+            "ShippingSpeedCategory": shipping_speed_category,
+            "FulfillmentPolicy": fulfillment_policy,
+        }
+
+        data.update(utils.enumerate_keyed_param("Items.Item", items or []))
```
Comment (GriceTurrble) [inline mws/apis/outbound_shipments.py]: Referring to [MWS documentation](http://docs.developer.amazonservices.com/en_US/fba_outbound/FBAOutbound_CreateFulfillmentOrder.html), this key is expected to be `Items.member.X.key`. Should change the keyed param argument to `Items.member`.

**Comment by GriceTurrble** [comment]
Initial changes look good so far. :)

I'd like to restrict the scope of this PR to the `create_fulfillment_order` method for now. Reason being, there are several types of data structures - like Items, DeliveryWindow, Address, etc. - that are expected to be sent with the request, and expecting the user to send all of these as one massive method call can be overwhelming, especially if any one of those data types has some error or if we expect the user to know the exact key names to use.

As a starting point, refer to the methods in `apis.inbound_shipments` on develop branch that parse the structure of item dicts. This is one way we've tried to provide useful errors when the user submits data, before attempting to send requests to MWS that we know will fail ahead of time.

Ideally, these data structures can be codified as separate classes within `apis.outbound_shipments`. That would provide the best end-user experience, I think.

---

That all is optional, of course. We can make due with request methods that work and tests to cover them. :)

**Comment by tgehrs** [comment]
That makes sense! I (randomly) picked `apis.merchant_fulfillment` as my guide and I can see how the methods in `api.inbound_shipments` are preferable. I will put in some time soon on another PR with fancier validation. If there ends up being code that can be reused should that live in `utils.py`? For example parse_item_args looks like it could be reused in OutboundShipments with very little modification (or even a more generic key_config validator). Also are there any classes right now that have the data structures codified as separate classes?

**Comment by GriceTurrble** [comment]
> For example parse_item_args looks like it could be reused in OutboundShipments with very little modification (or even a more generic key_config validator).

Perhaps, but given the different keys available between [`InboundShipmentItem`](http://docs.developer.amazonservices.com/en_US/fba_inbound/FBAInbound_Datatypes.html#InboundShipmentItem) and [`CreateFulfillmentOrderItem`](http://docs.developer.amazonservices.com/en_US/fba_outbound/FBAOutbound_Datatypes.html#CreateFulfillmentOrderItem), and considering Amazon may choose to alter one or the other at any time, I think it best to keep them separated in their own API sections.

> Also are there any classes right now that have the data structures codified as separate classes?

Very little. There is a [ReportType enumeration](https://github.com/python-amazon-mws/python-amazon-mws/blob/0300af9a07d5362115a9fed3746825eae2752160/mws/apis/reports.py#L223) available that simply defines the names of different reports, as defined [here (MWS docs)](http://docs.developer.amazonservices.com/en_US/reports/Reports_ReportType.html).

My suggestion here to write classes for these data types has me thinking I'll write one or more for InboundShipments sometime soon (*after* I finally get around to approving the other active PR we have (sorry, folks!)).

**Comment by tgehrs** [comment]
Ok cool. Do those type of changes belong on a separate PR? Should I fill in the other methods for OutboundShipments on this one?

**Pull request merged**